lib/vector-core/src/stream/driver.rs

```diff
@@ -232,19 +102,7 @@ where
                 biased;
 
                 // One or more of our service calls have completed.
-                Some(acks) = in_flight.next(), if !in_flight.is_empty() => {
-                    for ack in acks {
-                        let (seq_num, ack_size): (SequenceNumber, usize) = ack;
-                        let request_id = seq_num.id();
-                        trace!(message = "Acknowledging service request.", request_id, ack_size);
-                        ack_tracker.mark_seq_num_complete(seq_num, ack_size);
-                    }
-
-                    if let Some(ack_depth) = ack_tracker.consume_ack_depth() {
-                        trace!(message = "Acking events.", ack_size = ack_depth);
-                        acker.ack(ack_depth.get());
-                    }
-                }
+                Some(_acks) = in_flight.next(), if !in_flight.is_empty() => {}
```

Comment:
Why the empty branch?

Reply:
This probably deserves some discussion of what's going on, and was actually one of the parts I most wanted your eyes on. AFAICT the `in_flight` queue just keeps track that there are requests still hanging around waiting to be acked. This branch pops acks off of that queue, to be used in other branches to determine if there are requests in flight. The semi-hidden logic of that select is that if there are no requests in flight, no next batch, and no next input batch, then break the loop and exit.

Again, this is all based on my reading of your code there. I think we need to keep the `in_flight` queue so it can be waited on in that select, but the ack responses in the queue are all the null `()` type and we don't actually do anything with them.

Reply:
OK, yeah, this makes total sense now that I've taken some time to re-read through it.

I guess, maybe, you could change `Some(_acks)` to `_` or something just to hide a little more of the semi-convoluted reason of how these futures are being driven. Alternatively, we could change `FuturesUnorderedChunked` -- either by focusing it on this use case, or adding a specific impl for futures that return `()` -- such that it has a method called `drive` (or something like that) which is solely there to poll the inner `FuturesUnordered` until it has no more completed futures, and does not actually return a `Vec<T>`, since we have nothing to return to the caller in this case.

Up to you, since the code works as-is.

Reply:
Done in 39e92af48